mantidproject/mantid
# Apply placzek correction to polaris workflow

This PR add the calculation and subtraction of the placzek self scattering correction to the POLARIS scripts.
this correction is performed in the `create_total_scattering_pdf` method of Polaris. if `merge_banks=True` then the correction is applied before the detector banks are merged.

**To test:**
To script below will run the relivant methods in `Polaris`, as with the merging banks `create_total_scattering_pdf` can take a .lim file for merging paramiters, however the correction in this script applies to both `merge_banks` True and False. After being run with `merge_banks=True` the Function should output three things: a workspace contrianing the merged corrected spectra, and group workspace containing a single spectra workspace for each detector bank, and a workspace containing the PDF. Configuration files are needed to run this script, message me on slack and I'll send them over.

```
# The following line helps with future compatibility with Python 3
# print must now be used as a function, e.g print('Hello','World')
from __future__ import (absolute_import, division, print_function, unicode_literals)

# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *

import numpy as np

from isis_powder import polaris, SampleDetails
config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="test", mode="PDF")  

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si')
polaris.set_sample_details(sample=sample_details)

polaris.create_vanadium(first_cycle_run_no="98532",
                        multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = np.array([2.5, 3, 4, 6, 7, 3.5, 5, 7, 11, 40]).reshape((2, 5))
#q_lim = 'C:/Users/wey38795/Documents/gem61910.lim'
polaris.create_total_scattering_pdf(run_number="98533", q_lims=q_lim, merge_banks=True)
```

Fixes #26986 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
